### PR TITLE
Bay ports, fixes and stuff

### PR DESCRIPTION
**Pull request by Yoshax**
Putting it up so folks can see and work on it from here. Still got a bunch more stuff to grab.

Ports Baystation12/Baystation12#11281
Ports Baystation12/Baystation12#11988
Ports Baystation12/Baystation12#11507
Ports Baystation12/Baystation12#12683
Ports Baystation12/Baystation12#11431

### CODE REVIEW

File: code/game/antagonist/antagonist.dm [hunk 1]
```diff
@@ -116,46 +117,57 @@
...
+	if(active_antags >= cur_max)
+		log_debug("Could not auto-spawn a [role_text], active antag limit reached.")
+		return 0
+
+	build_candidate_list(flags & (ANTAG_OVERRIDE_MOB|ANTAG_OVERRIDE_JOB))
+	if(!candidates.len)
+		log_debug("Could not auto-spawn a [role_text], no candidates found.")
+		return 0
+
+	attempt_spawn(1) //auto-spawn antags one at a time
```
Comment (mwerezak) [inline code/game/antagonist/antagonist.dm]: This *really* doesn't look like it should be there. Bad merge with something Polaris added?

Comment (mwerezak) [inline code/game/antagonist/antagonist.dm]: Oh nvm I am just super tired

**Comment by Yoshax** [comment]
This is... Functional, I guess.

This does not:
- Ports Psi's scheduler stuff
- Adds the relevant unit tests
- Use the event system anywhere Psi has made it use it to clean stuff up

In addition, the `task_triggered.dm` is not ticked because it doesn't compile because of missing the scheduler stuff.

Trying to port all of that is suffering and I'm not going to. There's possibly a few other things I'm going to include, but this is essentially done.

**Comment by Yoshax** [comment]
Jk. It doesn't actually work. If someone cares, they can do this, I'm not.

**Pull request closed without merging**